packages/mui-material/src/Alert/Alert.js

```diff
@@ -196,15 +201,16 @@ const Alert = React.forwardRef(function Alert(inProps, ref) {
       ) : null}
       {action == null && onClose ? (
         <AlertAction ownerState={ownerState} className={classes.action}>
-          <IconButton
+          <AlertCloseButton
             size="small"
             aria-label={closeText}
             title={closeText}
             color="inherit"
             onClick={onClose}
+            {...componentsProps.closeButton}
```

Comment:
In the future we may want to allow here callbacks here as it is done in @mui/base, but for now it looks good 👍